### PR TITLE
feat: support agent config file

### PR DESCRIPTION
**Pull request by RedbeanGit**
+ Copy agent config file into archive
* Automatically set right mode for init, agent and config

### CODE REVIEW

File: initramfs/src/image.rs [hunk 1]
```diff
@@ -126,30 +130,35 @@ impl Image {
...
-                Builder::new("init").mode(init_mode),
-                Cursor::new(init_content),
-            ),
+            (Builder::new("init").mode(33277), Cursor::new(init_content)),
         );
         entries.insert(
             "agent".to_string(),
             (
-                Builder::new("agent").mode(agent_mode),
+                Builder::new("agent").mode(33277),
```
Comment (sameo) [inline initramfs/src/image.rs]: What is 33277?

File: initramfs/src/image.rs [hunk 2]
```diff
@@ -126,30 +130,35 @@ impl Image {
...
                 Cursor::new(agent_content),
             ),
         );
+        entries.insert(
+            "config.yaml".to_string(),
+            (
+                Builder::new("config.yaml").mode(33204),
```
Comment (sameo) [inline initramfs/src/image.rs]: Same for 33204. You should define a constant for those, with a relevant name.

File: initramfs/src/image.rs (hunk 1 above)
Comment (RedbeanGit) [reply initramfs/src/image.rs]: Hmmm I think I should use the octal version. 33277 in decimal is equivalent to 100775 (a regular file with 0 for special bit, rwx permissions for owner and group and rx for others)

File: initramfs/src/image.rs (hunk 2 above)
Comment (RedbeanGit) [reply initramfs/src/image.rs]: Same here. I'll create a constant

**Review by sameo — APPROVED** [review]

**Pull request merged**